### PR TITLE
Fix a node with an invalid number of outputs

### PR DESCRIPTION
**Pull request by GogoVega**
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Proposed changes

Fix a node with an invalid number of outputs - isNaN check should be at the top.

Retrieve `nodeImported.outputs` only if the node definition allows it.

Related PR: #4757.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `npm run test` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by knolleary** [comment]
Can you provide an example that will demonstrate the bug this is fixing? Given we have never had an issue raised about this bit of code, it's hard to see what it's fixing.

You are adding extra conditionals to some of the if statements and I can't see why. 
```
if (n.hasOwnProperty('inputs') && def.defaults.hasOwnProperty("inputs")) {
```

**Comment by GogoVega** [comment]
Of course, if you import this node, you will have a lot of errors because the `isNaN` has not been checked.

```json
[{"id":"d2fea84b9b1de13b","type":"function","z":"15cdbf1d258ebb57","name":"function 1","func":"\nreturn msg;","outputs":"badNumber","timeout":0,"noerr":0,"initialize":"","finalize":"","libs":[],"x":480,"y":540,"wires":[[]]}]
```

And for the second, this node should have one outputs.

```json
[{"id":"eead240a9748858b","type":"range","z":"15cdbf1d258ebb57","outputs":3,"minin":"","maxin":"","minout":"","maxout":"","action":"scale","round":false,"property":"payload","name":"","x":420,"y":540,"wires":[[],[],[]]}]
```

**Review by knolleary — APPROVED** [review]

File: packages/node_modules/@node-red/editor-client/src/js/nodes.js [hunk 1]
```diff
@@ -2321,29 +2321,31 @@ RED.nodes = (function() {
...
-                                    // If 'wires' is longer than outputs, clip wires
-                                    console.log("Warning: node.wires longer than node.outputs - trimming wires:",node.id," wires:",node.wires.length," outputs:",node.outputs);
-                                    node.wires = node.wires.slice(0,node.outputs);
-                                } else {
-                                    // The node declares outputs in its defaults, but has not got a valid value
-                                    // Defer to the length of the wires array
+
+                            // The node declares outputs in its defaults, but has not got a valid value
+                            // Defer to the length of the wires array
+                            if (node.hasOwnProperty('wires'))
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/nodes.js:2339]:
```suggestion
                            if (node.hasOwnProperty('wires')) {
```

Comment (GogoVega) [reply packages/node_modules/@node-red/editor-client/src/js/nodes.js]: Woops

**Pull request merged**